### PR TITLE
Update theme.inc

### LINKED ISSUE #53: One more theme.inc change
Hi, 

One last theme.inc change that reduced errors on my site.
[theme_inc.zip](https://github.com/d6lts/drupal/files/4948198/theme_inc.zip)

### PR DESCRIPTION
**Pull request by NiklasBr**
Resolves #53

### CODE REVIEW

File: includes/theme.inc [hunk 1]
```diff
@@ -1216,7 +1216,7 @@ function theme_links($links, $attributes = array('class' => 'links')) {
   global $language;
   $output = '';
 
-  if (count($links) > 0) {
+  if (!empty($links)) {
```
Comment (dsnopek) [inline includes/theme.inc]: I'm not sure this is the right change. The `count()` will cause problems if `$links` is not an array, but only a few lines later, there's:

```
$num_links = count($links);
```

... which could cause problems again, if the `$links` is not empty, but still not an array.

So, perhaps the change should really be:

```
if (is_array($links) && count($links) > 0) {
```

**Review by dsnopek — CHANGES_REQUESTED** [review]

File: includes/theme.inc (hunk 1 above)
Comment (NiklasBr) [reply includes/theme.inc]: `empty()` should work fine, see this example:

```php
$str_empty = '';
$str_content = 'string';
$array_empty = [];
$array_content = ['array'];
$null = null;

var_dump([
    empty($str_empty),
    empty($str_content),
    empty($array_empty),
    empty($array_content),
    empty($null),
    empty($var_does_not_exist)
]);
```

Output is the same with PHP 5.6 and 7.3:
```php
array(6) {
  [0]=>bool(true)
  [1]=>bool(false)
  [2]=>bool(true)
  [3]=>bool(false)
  [4]=>bool(true)
  [5]=>bool(true)
}
```

Comment (dsnopek) [reply includes/theme.inc]: Yes, but 3 lines later the code uses `count()` again. I suppose you could have replaced both `count()`'s with `!empty()`? In any case, the new version of this PR works for me!

Comment (splash112) [reply includes/theme.inc]: Lines like "if (count($links) > 0)" were used back in the days to circumvent some problems with empty() in ancient PHP versions. Those problems have been fixed and errors introduced with using count(), so moving to empty() should not be a problem. In instances that you use count to actually count something, don't.

**Comment by dsnopek** [comment]
Thanks!

**Pull request merged**